system/jlib/jptree.cpp

```diff
@@ -4093,7 +4093,15 @@ class CommonReaderBase : public CInterface
     }
     inline void skipWS()
     {
-        while (isspace(nextChar)) readNext();
+        while (isspace(nextChar) || nextChar=='#')
+        {
+            if (nextChar=='#')
+            {
+                while (nextChar != '\n')
```

Comment:
need to cope with # on the end of the file with no \n (and osx files that only end in \r)